src/manage_pg.c

```diff
@@ -307,10 +307,43 @@ manage_create_sql_functions ()
        GVM_LIB_INSTALL_DIR);
 
   sql ("CREATE OR REPLACE FUNCTION valid_db_resource_type (text)"
```

Comment:
Here also, it would be good to make it clear that it must match the C valid_db_resource_type.

Even better would be to have them defined in only one place, maybe using macros, but I guess that gets a bit complex.

Reply:
Yes, it must be in sync with the C code. I already thought about it and my idea was to use only the database ones. Since there are not as many calls to these three functions it would be less effort than writing macros. What do you think? Do you see performance impacts?

Reply:
Looking a bit closer at valid_db_resource_type, the SQL function is only used by the SQL resource_name.  The SQL resource_name is only used by the permission iterator.  I think we should simply drop the call to valid_resource_type in SQL resource_name, and remove the function.  It's unlikely (maybe even impossible) that we get an invalid resource type in SQL resource_name.  And if we do it will just be a SQL error because the SELECT will use an invalid table.

Reply:
About always using SQL for level_max/min_severity, these are used by severity_data_level_counts, which is used by report_counts_id_full.  Result counting is a performance hotspot, but severity_data_level_counts is only called once per report so I think it will not have a significant perfomance impact.

However, it's ugly, and does not fit in line with reducing the number of queries to the db.

I think duplicating the code is the neater option, but will not block you if you prefer using an SQL function.

Reply:
And to be clear, I"m also happy with just duplicating the code (with a clear comment), to avoid complex macros.

Reply:
I just commited the suggested changes. Are the comments clear enough?

By the way, I understand that you want to keep the queries to the db as low as possible!

Reply:
Yes, that's clear, thank you.